asapdiscovery-ml/asapdiscovery/ml/cli_mlops.py

```diff
@@ -642,20 +828,56 @@ def train_GAT_for_endpoint(
-    # now push weights, config and manifest to S3
-    logger.info("Pushing weights, config and manifest to S3")
-    # push the whole final directory to S3
-    # ends up at BUCKET_NAME/protocol/model_tag
-    s3 = S3.from_settings(s3_settings)
-    s3_ensemble_dest = f"{protocol}/{model_tag}"
-
-    # push ensemble to "latest"
+    # need for both test and not test
     s3_manifest_dest = f"{protocol}/latest/manifest.yaml"
 
-    logger.info(f"Pushing final directory to S3 at {s3_ensemble_dest}")
-    s3.push_dir(final_dir_path, s3_ensemble_dest)
-
-    logger.info(f"Pushing manifest to S3 at {s3_manifest_dest}")
-    s3.push_file(manifest_path, s3_manifest_dest)
-
-    logger.info("Done.")
+    if test:
+        logger.info("Test mode, not pushing to S3")
+    else:
+        # now push weights, config and manifest to S3
+        logger.info("Pushing weights, config and manifest to S3")
+        # push the whole final directory to S3
+        # ends up at BUCKET_NAME/protocol/model_tag
+        s3 = S3.from_settings(s3_settings)
+        s3_ensemble_dest = f"{protocol}/{model_tag}"
+
+        # push ensemble to "latest"
+        logger.info(f"Pushing final directory to S3 at {s3_ensemble_dest}")
+        s3.push_dir(final_dir_path, s3_ensemble_dest)
+
+        logger.info(f"Pushing manifest to S3 at {s3_manifest_dest}")
+        s3.push_file(manifest_path, s3_manifest_dest)
+
+    logger.info("Done pushing")
+    # grab test CSV for plotting
+    ds_test_end_path = output_dir / "ds_test.csv"
+    shutil.copy(test_csv_path, ds_test_end_path)
+    logger.info(f"Test CSV copied from  {test_csv_path} to {ds_test_end_path}")
+    # use actual manifest URL for the next step, not just our push location, if testing will pull down older manifest
+    manifest_url = f"{SKYNET_SERVE_URL}/{s3_manifest_dest}"
+    logger.info(f"Manifest URL: {manifest_url}")
+    # use the manifest we just made to make a new manifest for the latest ensemble
+    logger.info("Plotting test performance for ensemble over test set")
+    ens_models = RemoteEnsembleHelper(manifest_url=manifest_url).to_ensemble_spec()
+    # dict with one item, grab the model
+    model = list(ens_models.values())[0]
+    logger.info(f"Model: {model}")
+    plot_path = plot_test_performance(ds_test_end_path, readout, model, output_dir)
+    logger.info(f"Test performance plot saved to {plot_path}")
+
+    if test:
+        logger.info("Test mode, not pushing to S3")
+    else:
+        # push the plot to S3 in the same location as the ensemble
+        s3_plot_ens_dest = f"{protocol}/{model_tag}/test_performance.png"
+        logger.info(f"Pushing test performance plot to S3 at {s3_plot_ens_dest}")
+        s3.push_file(plot_path, s3_plot_ens_dest)
+
+    # log the artifact to WandB run
+    logger.info("Logging test performance plot to WandB")
+    for run_id in wandb_run_ids:
+        if run_id is None:
+            continue
+        wandb.init(id=run_id, project=wandb_project, resume="must")
```

Comment:
this can raise an error I think if the W&B specification doesn't match an existing run (I think `wandb.errors.UsageError`), can you add a quick `try...except` block to handle that?